src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HeartbeatDefaultPayload.cs

```diff
@@ -0,0 +1,94 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+
+    internal static class HeartbeatDefaultPayload
+    {
+       public static readonly string[] DefaultFields =
+        {
+            "runtimeFramework",
+            "baseSdkTargetFramework"
+        };
+
+        public static void GetPayloadProperties(IEnumerable<string> disabledFields, IHeartbeatProvider provider)
+        {
+            var enabledProperties = HeartbeatDefaultPayload.RemoveDisabledDefaultFields(disabledFields);
+
+            var payload = new Dictionary<string, HeartbeatPropertyPayload>();
+            foreach (string fieldName in enabledProperties)
+            {
+                try
+                {
+                    switch (fieldName)
+                    {
+                        case "runtimeFramework":
+                            provider.AddHealthProperty(fieldName, HeartbeatDefaultPayload.GetRuntimeFrameworkVer(), true);
+                            break;
+                        case "baseSdkTargetFramework":
+                            provider.AddHealthProperty(fieldName, HeartbeatDefaultPayload.GetBaseSdkTargetFramework(), true);
+                            break;
+                        default:
+                            provider.AddHealthProperty(fieldName, "UNDEFINED", false);
+                            break;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    CoreEventSource.Log.FailedToObtainDefaultHeartbeatProperty(fieldName, ex.ToString());
+                }
+            }
+        }
+
+        private static List<string> RemoveDisabledDefaultFields(IEnumerable<string> disabledFields)
+        {
+            List<string> enabledProperties = new List<string>();
+
+            if (disabledFields == null || disabledFields.Count() <= 0)
+            {
+                enabledProperties = DefaultFields.ToList();
+            }
+            else
+            {
+                enabledProperties = new List<string>();
+                foreach (string fieldName in DefaultFields)
+                {
+                    if (!disabledFields.Contains(fieldName, StringComparer.OrdinalIgnoreCase))
+                    {
+                        enabledProperties.Add(fieldName);
+                    }
+                }
+            }
+
+            return enabledProperties;
+        }
+
+        private static string GetBaseSdkTargetFramework()
+        {
+#if NET45
+            return "net45";
+#elif NET46
+            return "net46";
+#elif NETSTANDARD1_3
+            return "netstandard1.3";
+#else
+#error Unrecognized framework
+            return "undefined";
+#endif
+        }
+
+        private static string GetRuntimeFrameworkVer()
```

Comment:
Yes, I am working on (and sitting on) a PR that covers only the SDK default payload. We can get to that once this PR is complete.

Reply:
If there is a separate PR coming for default payload, why not remove it from this one to avoid inconsistency later? Keep basesdktargetframeowkr, but runtimeframework can be removed if its not finished work.

Reply:
Runtime Framework is *basically* finished here. It works and gives us the current version of the runtime framework, just not the exact moniker for ".NET Core 2.0" or simple things like this. I'm not confident we will ever get it to that level of detail anyways, since there is not a real way to get that information across .NET frameworks nor versions (even the cli 'dotnet' doesn't have a generic way of obtaining this information from what I've seen).

Reply:
without correctly identifying runtime framework including .net core vs .net framework, i don't think this field is very useful in customer support scenarios where one of the basic question is asking 'is it .net core or .net framework' and 'what version'.

I believe i mentioned this in an earlier commment, but web sdk has code which determines the version of .net core runtime. It does this by looking at the version of httpclient, which changed between .net core 1.1 and 2.0. 

This is for identifying which .NET Framework is;
https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/develop/Test/DependencyCollector_deprecated/FunctionalTests/FuncTest/Helpers/DotNetVersionCheck.cs

for .net core, identifying if .net 2.0 or not.
https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/bcdac29fcaa3f3178c4bf0f83375b42103c8bfa8/Src/DependencyCollector/Shared/Implementation/HttpCoreDiagnosticSourceListener.cs#L73

It maybe tricky to do this for all versions, so my recommendation is to remove this field until we can reliably identify atleast .net core vs .net framework.

Reply:
I'm still searching for how to obtain this information in a robust & generic way that won't fail when a new framework will be released.

My current understanding is that the target framework we build for tells us .NET Core/.NET Standard, and the current framework is obtained via the method I've written thus far (and one other that is available in .NET Core). Those two items together tell us the framework/runtime I think, at least enough to allow us to deduce the .NET Core/.NET Framework question. I am totally fine with being wrong here, I would love to know specifics though.